galasa-eclipse-parent/dev.galasa.eclipse/src/dev/galasa/eclipse/ui/run/storedartifacts/ImageView.java

```diff
@@ -0,0 +1,138 @@
+package dev.galasa.eclipse.ui.run.storedartifacts;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.UUID;
+
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.action.IToolBarManager;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.PaintEvent;
+import org.eclipse.swt.events.PaintListener;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.widgets.Canvas;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.part.ViewPart;
+
+import dev.galasa.eclipse.Activator;
+
+public class ImageView extends ViewPart implements PaintListener {
+	
+	 public static final String  ID = "dev.galasa.eclipse.ui.run.storedartifacts.ImageView";
+
+	 private Path                path;
+	 private Canvas              canvas;
+	 private Composite 			 parent;
+	 private Action 			 saveImageToWorkspace;
+	 
+	 private boolean 			 cached = false;
+	 private Path				 cachedPath;
+	 
+	 private Image				 image;
+	 
+	@Override
+	public void createPartControl(Composite parent) {
+        this.canvas = new Canvas(parent, SWT.NULL);
+        this.parent = parent;
+        canvas.addPaintListener(this);		
+        
+        createSaveAction();
+        createToolbar();
+	}
+
+	@Override
+	public void setFocus() {
+		this.canvas.setFocus();
+		
+	}
+
+	@Override
+	public void paintControl(PaintEvent event) {
+		try {
+			if (!cached) {
+				String fileName = this.path.getFileName().toString();
+				cachedPath = Activator.getCachePath().resolve(fileName);
+				Files.copy(this.path, cachedPath);
+				
+				InputStream is = Files.newInputStream(this.cachedPath);
```

Comment:
Can't do this on the UI Thread.  if this is pulling from the CouchDB RAS it will lock up the UI thread.

Reply:
you are going to have to pull the image using a Job.